lib/application.js

```diff
@@ -278,12 +279,40 @@ Origin.prototype.createServer = function (options, cb) {
         requestDomain.on('error', next);
         requestDomain.run(next);
       });
+
+
+ 
+
       server.use(auth.initialize());
       server.use(auth.session());
       server.use(express.static(path.join(app.configuration.serverRoot, 'frontend', 'src')));
       server.use(permissions.policyChecker());
-      server.use(server.router);
-      server.use(app.clientErrorHandler());
+        // loop through the plugins for middleware
+        var pluginManager = pluginmanager.getManager();
+        var plugins = pluginManager.getPlugins();
+        var plugin;
+        async.forEach(Object.keys(plugins), function (directory, cbOuter){ 
```

Comment:
since this process is only run once at server startup, you could use Array.forEach without worrying about blocking. It might help make this a bit more readable.